devel/ci/integration/bodhi/start.sh

```diff
@@ -1,8 +1,16 @@
 set -m
 
 # Bodhi backend
+echo "Starting Fedora Messaging consumers"
 fedora-messaging consume &
 
+# Celery
+echo "Starting Celery"
+pushd /  # Otherwise celeryconfig will be picked up from CWD (/bodhi)
```

Comment:
I haven't found any. It's an unfortunate but common problem with apps that choose to use a python module for their configuration instead of a regular file: you're dependant on Python's import paths. If there's a better way, I'm all ears, though.

Reply:
We could use a custom module finder/loader such as I mentioned when we talked before, it would let us/celery import configuration from a defined module name but we get to choose the file that actually gets loaded by e.g. an absolute path in the configuration, like this (untested):

* in `bodhi/server/util.py`:

```
...
class PyFileAsModule:
    """Module finder/loader for arbitrary Python files."""

    def __init__(self, filename: str, modname: str):
        if modname in sys.modules:
            raise ValueError(f"Module {modname} exists in sys.modules.")

        if modname in sys.meta_path:
            raise ValueError(f"Module finder for {modname} registered in sys.meta_path.")

        self.filename = filename
        self.modname = modname

    @classmethod
    def register(cls, filename: str, modname: str) -> str:
        """Create and register a module finder/loader for Python files."""
        obj = cls(filename=filename, modname=modname)
        sys.meta_path.append(obj)
        return modname

    def find_module(self,
                    fullname: str,
                    path: str = None) -> typing.Optional['PyFileAsModule']:
        """Return self as the loader if responsible."""
        if fullname == self.modname:
            return self

    def load_module(self, fullname: str) -> types.ModuleType:
        """Return a module from a registered file."""
        if fullname in sys.modules:
            return sys.modules[fullname]

        try:
            m = types.ModuleType(fullname)
            m.__file__ = self.filename
            m.__loader__ = self
            sys.modules[fullname] = m

            with open(self.filename, 'r') as f:
                exec(f, sys.modules[fullname].__dict__)

            return m
        except Exception as e:
            if fullname in sys.modules:
                del sys.modules[fullname]
            raise e
```

* in `bodhi/server/tasks/__init__.py`:

```
...
from bodhi.server.util import PyFileAsModule
...
# The Celery app object.
app = celery.Celery()
app.config_from_object(PyFileAsModule.register(config["celery_config"], 'celeryconfig'))
...
```

Reply:
Hmm, it seems Celery's way to find its configuration is broken enough for the above -- even if fixed -- not to work.